### PR TITLE
Bring man pages in-tree

### PR DESCRIPTION
**Pull request by dave-pi**
Attempt to bring man pages in tree from gnuradio/pkg-gnuradio.

- Created docs/man with man pages
- Changed cmake to integrate MAN dir, build and install
- Added a sed to replace VERSION in man pages
- Started registering man pages as a GrComponent (similar to doxgen)

### CODE REVIEW

**Comment by dave-pi** [comment]
> I think it is way overkill to use sed and to compress the man pages.
I added compression mostly because almost all man pages in my system seem to be compressed (see below for rest of discussion).

> CMake provides configure_file() that already works cross-platform.
Indeed. It does seem a better option than sed, as it's cleaner and cross-platform. Also, I changed the strings to be of the form @VERSION@ instead of ${VERSION} since that is what Doxyfile.in is using. The only caveat is that configure_file() is a configure-time command and not a build-time command. So for example, if I run cmake and then remove a build/docs/man/<man_page>, then when I run make, the bzip fails because the man page is not there and it doesn't know how to generate it because there's no rule to build it (since it was copied at configure time). Doxygen has the same problem. It's a corner case so I'm fine going ahead with it.

> Compression is a packaging thing, handled in the src.rpm spec file or
> Debian tools like dh_installman.
That sounds about right, I'm just thinking about people building from source. I guess we can just assume building from source is a developer thing and they couldn't care less about compressing man pages. If that's the case I'm happy to remove it altogether.

> At least, compression should be optional. So no need for having
> a CMake FATAL_ERROR - embedding folk will hate that.
Fatal definitely seemed a bit harsh so I made it optional (for now, until we decide if we keep it optional or remove it).

**Comment by dave-pi** [comment]
So given the absence of further feedback here is the proposal:

- Put man pages in man/docs and give them a .in extension
- Replace version and date placeholders at cmake configure time. For
  that to work moved date generation to top-level CMakeLists.txt
- Improve gnuradio-companion man page, expanding description and
  updating options
- Remove help2man generated man pages, and man pages for non-existent
  commands

Content-wise the man pages need further work but I think this is a good checkpoint to avoid making this change too convoluted.

**Comment by dave-pi** [comment]
> This overlaps with another previous pull request:
> cmake: use TIMESTAMP function to generate build_date and save to cache
> #3547
> 
> I think that patch should be merged first both to master and maint-3.8, then reduce this PR to just the man pages.

Sure that makes sense. I'll change this around to assume the variables are already available.

**Comment by dave-pi** [comment]
I've uploaded a new commit that restricts the changes to the man pages. Notice I'm using a @BUILD_DATE_SHORT@ because the default one I think is too verbose to include in a man page. So after #3547 goes in, this change might need a further tweak.

I've tested enabling and disabling man-pages component and the configure seems to work. I'll try make install just to see if the man pages land in the right place and there are no wrinkles.

File: docs/man/CMakeLists.txt
Comment (marcusmueller) [inline docs/man/CMakeLists.txt:1]: 2020, and unless you want to, you don't have to assign copyright to the FSF anymore (but you'll need to sign-off (`git commit -s`) on your commits)

File: docs/man/dial_tone.1.in
Comment (mbr0wn) [inline docs/man/dial_tone.1.in:1]: Did you write these? Or take them from Maitland's repo?

Comment (mbr0wn) [reply docs/man/dial_tone.1.in:1]: ...you wrote that in the description. Move on...

**Review by mbr0wn — APPROVED** [review]
Looks fine.

**Comment by dave-pi** [comment]
Just to remind this change uses a @BUILD_DATE_SHORT@ which doesn't exist yet. So I'm waiting for #3547 to go in first (which refactors the build date generation logic) to add it. Maybe add a 'don't merge' tag.

**Comment by mbr0wn** [comment]
> Just to remind this change uses a @BUILD_DATE_SHORT@ which doesn't exist yet. So I'm waiting for #3547 to go in first (which refactors the build date generation logic) to add it. Maybe add a 'don't merge' tag.

Yeah I've been chatting with the reviewers and authors on that PR to see if we can merge that.

**Comment by mbr0wn** [comment]
> Just to remind this change uses a @BUILD_DATE_SHORT@ which doesn't exist yet. So I'm waiting for #3547 to go in first (which refactors the build date generation logic) to add it. Maybe add a 'don't merge' tag.

OK, it's merged.

**Comment by mbr0wn** [comment]
@dave-pi please remove the "Don't Merge" tag when ready.

**Comment by dave-pi** [comment]
Think it's ready now. Tested build and seems to work. Note I moved BUILD_DATE generation to top CMakeLists.txt so it's available in other modules (not sure why it had been moved to gnuradio-runtime/lib/CMakeLists.txt). constants.cc generation seems to work fine so should be OK.
Also added Z suffix to BUILD_DATE to indicate UTC.

**Comment by mbr0wn** [comment]
OK looks like no one is complaining here. It's going in.

**Comment by mbr0wn** [comment]
@maitbot FYI that I merged this. Will this have effects no your repo?

**Pull request merged**
